commit c596bb2c28271ba1ba0b6af4ef4a3244b32bbfe1 (refs/changes/54/3954/5)
Author: Jason King <jason.king@joyent.com>
Date:   2018-05-15T12:29:41-05:00 (1 year, 5 months ago)
    
    OS-6935 Possible race in lx_start_nfs_lockd
    Reviewed by: Jerry Jelinek <jerry.jelinek@joyent.com>
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/os/lx_lockd.c b/usr/src/uts/common/brand/lx/os/lx_lockd.c
index 234e815d3f..d6d965398a 100644
--- a/usr/src/uts/common/brand/lx/os/lx_lockd.c
+++ b/usr/src/uts/common/brand/lx/os/lx_lockd.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2017 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -77,6 +77,11 @@ lx_lockd_alive(pid_t lockd_pid)
 	}
 
 	mutex_enter(&p->p_lock);
+	if (p->p_stat == SZOMB || (p->p_flag & SEXITING) != 0) {
+		mutex_exit(&p->p_lock);
+		mutex_exit(&pidlock);
+		return (B_FALSE);
+	}
 	vp = p->p_exec;
 	VN_HOLD(vp);
 	mutex_exit(&p->p_lock);
